### DISCUSSION TITLE
Editor selection box should be anchored in time in scrolling rulesets

### DISCUSSION THREAD

**Discussion by ekrctb** [ppy/osu#18148] [Ideas]
For selection in the timeline, a selection box remembers the time it started the selection after the current time is changed.
This behavior should be applied to a selection box in the composer as well for scrolling rulesets (taiko, catch, mania).

In the osu!catch editor, suppose there is a hit object at `time = 100, x = 128`. The current time is `0`.

1. Start a selection in the timeline by mouse down at `time = 0` and `x = 0`.
2. Change the current time to `200` by mouse scroll. Now, the cursor is not moved but the time at the cursor is changed to `200`.
3. Move mouse horizontally to `x = 256`.
4. End the selection by mouse up.

Expected behavior: the hit object at `time = 100` is selected.
Actual behavior: the hit object is not selected because the selection box height is zero.

**Comment by peppy** [comment]
Just to confirm, the case works okay if you make a larger selection rectangle right? The issue is when the "fixed" selection box doesn't encompass objects which have occurred in the traversed time span.

ie. working (ignoring how broken the selection box looks visually):

https://user-images.githubusercontent.com/191335/167397126-f2e8ad6e-dfbd-4884-8e3c-e238eaa716d1.mp4

**Reply by ekrctb** [reply]
The behavior is different when autoscrolling (play/pause) is used rather than mouse scrolling. I didn't found yet if that is intentional.

**Reply by peppy** [reply]
For a scrolling ruleset, I think both of those behaviours should match.

The reasoning for it not matching is for osu!, for instance, where a drag selection is generally operating in positional axes only rather than time (where it maybe makes more sense to only select on-screen objects when the track isn't playing).

That said, I wouldn't be against simplifying the behaviour to *always* add to selection chronologically while a drag is active and the track is seeked.
